### ISSUE TITLE
False positive for `Lint/ToEnumArguments`

### ISSUE THREAD

**Issue by marocchino** [rubocop/rubocop#12746]
It seems a bit odd to have the call pass all arguments even though there are no arguments.

## Expected behavior

No Offence

## Actual behavior
```
$ echo "def m; to_enum(:not_m); end" | bundle exec rubocop --stdin test.rb --only Lint/ToEnumArguments
Inspecting 1 file
W

Offenses:

test.rb:1:8: W: Lint/ToEnumArguments: Ensure you correctly provided all the arguments.
def m; to_enum(:not_m); end
       ^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

see Actual behavior

## RuboCop version

```
$ rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.1.3) [arm64-darwin22]
  - rubocop-capybara 2.20.0
  - rubocop-factory_bot 2.25.1
  - rubocop-performance 1.20.2
  - rubocop-rails 2.23.1
  - rubocop-rspec 2.26.1
```
